### PR TITLE
fix: directories with .sol in the name path treated as files

### PR DESCRIPTION
**Pull request by dbale-altoros**
This PR adds E2E tests for the pr #352 
Credits to @pycckuu

### CODE REVIEW

File: e2e/04-dotSol-on-path/contracts/ERC20.sol/Foo.sol.sol [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pragma solidity >=0.5.0;
```
Comment (fvictorio) [inline e2e/04-dotSol-on-path/contracts/ERC20.sol/Foo.sol.sol:1]: `Foo.sol.sol`?

File: e2e/test.js [hunk 2]
```diff
@@ -111,4 +111,16 @@ describe('e2e', function() {
       expect(stdout.trim()).to.contain('Code contains empty blocks')
     })
   })
+
+  describe('.sol on path', function() {
+    useFixture('04-dotSol-on-path')
+
+    it('should success', function() {
```
Comment (fvictorio) [inline e2e/test.js:118]:
```suggestion
    it('should handle directory names that end with .sol', function() {
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,12 +1,24 @@
+## [3.3.9] - TBD
```
Comment (fvictorio) [inline CHANGELOG.md:1]: This change doesn't seem to belong to this PR.

File: e2e/test.js [hunk 4]
```diff
@@ -111,4 +111,16 @@ describe('e2e', function() {
...
+      const { code, stdout } = shell.exec('solhint contracts/**/*.sol')
+
+      expect(code).to.equal(0)
+
+      console.log(stdout)
```
Comment (fvictorio) [inline e2e/test.js:123]: This console.log doesn't seem necessary.

File: lib/index.js [hunk 5]
```diff
@@ -61,7 +61,7 @@ function processFile(file, config) {
 function processPath(path, config) {
   const ignoreFilter = ignore({ allowRelativePaths: true }).add(config.excludedFiles)
 
-  const allFiles = glob.sync(path, {})
+  const allFiles = glob.sync(path, {}).filter((f) => fs.lstatSync(f).isFile())
```
Comment (fvictorio) [inline lib/index.js:64]: `glob` has a `nodir` option that we can use instead of doing this.

**Review by fvictorio — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (dbale-altoros) [reply CHANGELOG.md:1]: I corrected the old changelog and added this pr as well
The point is to have it up to date and update it on every change after this one

**Comment by dbale-altoros** [comment]
@fvictorio 
I corrected all issues you requested, some of them belong to the original coder.
The changelog is modified because I added this fix and also corrected miss information of previous version
Do you think it can be approved so we can move on ?
Thanks!

**Review by fvictorio — APPROVED** [review]
